test/unit/mdc-menu/menu.foundation.test.js

```diff
@@ -80,1165 +51,431 @@ test('exports cssClasses', () => {
-    foundation.open();
-    mockRaf.flush();
-    td.verify(mockAdapter.setTransformOrigin('left bottom'));
-    td.verify(mockAdapter.setPosition({left: '7px', bottom: '15px'}));
-  });
-
-testFoundation('#open anchors the menu to the bottom left in RTL when close to the bottom and right edge with margin',
-  ({foundation, mockAdapter, mockRaf}) => {
-    initAnchorLayout(mockAdapter, smallBottomRight, true);
-    foundation.setAnchorCorner(Corner.BOTTOM_START);
-    foundation.setAnchorMargin({top: 5, bottom: 10, right: 7});
-    foundation.open();
-    mockRaf.flush();
-    td.verify(mockAdapter.setTransformOrigin('right bottom'));
-    td.verify(mockAdapter.setPosition({right: '7px', bottom: '15px'}));
-  });
-
-testFoundation('opening menu should automatically select the last selected item if rememberSelection is true',
-  ({foundation, mockAdapter, mockRaf}) => {
-    const handlers = captureHandlers(mockAdapter, 'registerInteractionHandler');
-    const clock = lolex.install();
-    const target = {};
-    const expectedIndex = 2;
-    td.when(mockAdapter.getIndexForEventTarget(target)).thenReturn(expectedIndex);
-    td.when(mockAdapter.getNumberOfItems()).thenReturn(3);
-
-    foundation.init();
-    foundation.setRememberSelection(true);
-    handlers.click({target});
-
-    clock.tick(numbers.SELECTED_TRIGGER_DELAY);
-    foundation.open();
-    mockRaf.flush();
-
-    td.verify(mockAdapter.focusItemAtIndex(expectedIndex), {times: 1});
-
-    clock.uninstall();
-  });
-
-testFoundation('#close does nothing if event target has aria-disabled set to true',
-  ({foundation, mockAdapter}) => {
-    const mockEvt = {
-      target: {},
-      stopPropagation: td.func('stopPropagation'),
-    };
-
-    td.when(mockAdapter.getAttributeForEventTarget(td.matchers.anything(), strings.ARIA_DISABLED_ATTR))
-      .thenReturn('true');
-
-    foundation.close(mockEvt);
-
-    td.verify(mockAdapter.deregisterBodyClickHandler(td.matchers.anything()), {times: 0});
-  });
-
-testFoundation('#close adds the animation class to start an animation', ({foundation, mockAdapter}) => {
-  foundation.close();
-  td.verify(mockAdapter.addClass(cssClasses.ANIMATING_CLOSED));
-});
+test('handleKeydown space/enter key causes the menu to close', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const key = {key: 'Space', target: 'My Element', preventDefault: td.func('preventDefault')};
+  td.when(mockAdapter.elementContainsClass(key.target, listClasses.LIST_ITEM_CLASS)).thenReturn(true);
+  td.when(mockAdapter.getElementIndex(key.target)).thenReturn(0);
 
-testFoundation('#close does not add animation class if quickOpen is set to true', ({foundation, mockAdapter}) => {
-  foundation.setQuickOpen(true);
-  foundation.close();
-  td.verify(mockAdapter.addClass(cssClasses.ANIMATING_CLOSED), {times: 0});
-  td.verify(mockAdapter.removeClass(cssClasses.ANIMATING_CLOSED), {times: 0});
-});
+  foundation.handleKeydown(key);
```

Comment:
Should this be running another test with the enter key and checking for 2 calls?

Ditto on the next few tests

Reply:
2 calls of the `closeSurface` ?

Reply:
In this test, yes; 2 calls of notifySelected in the next one, etc.

At least that was my idea, as the shortest way to test both keys.